ttnn/cpp/ttnn/operations/data_movement/unsqueeze/unsqueeze.cpp

```diff
@@ -31,11 +30,11 @@ ttnn::Tensor UnsqueezeOperation::invoke(const ttnn::Tensor& input_tensor, const
     }
 
     // If the dimension is at the end, append it
-    if (normal_dim >= tensor_shape.size()) {
+    if (normal_dim == rank) {
         output_shape_vector.push_back(1);
     }
 
-    return ttnn::reshape(input_tensor, ttnn::SimpleShape(std::move(output_shape_vector)));
+    return ttnn::reshape(input_tensor, ttnn::Shape(output_shape_vector));
```

Comment:
why do you need to switch to ttnn::Shape? My understanding that we move our code to use SimpleShape everywhere possible and in future remove ttnn::Shape and ttnn::LegacyShape

@ayerofieiev-tt ^^

Reply:
I wanted to avoid an extra function call unnecessarily. Reshape currently converts SimpleShape to Shape and then operates with Shape. So why make it SimpleShape to immediately convert it back to Shape? However reshape also has an invoke taking the small vector directly, so I am going to switch over to doing that instead, if in the future Shape is obsoleted then we would only need to modify reshape and not unsqueeze.

Reply:
That's a better approach. You should now try to avoid using Shape if possible.